sdk/src/client/api/block_builder/transaction_builder/requirement/context_inputs.rs

```diff
@@ -81,25 +83,16 @@ impl InputSelection {
             needs_commitment_context = true;
         }
         // BlockIssuanceCreditContextInput requires a CommitmentContextInput.
-        if self
-            .context_inputs
-            .iter()
-            .any(|c| c.kind() == BlockIssuanceCreditContextInput::KIND)
-        {
+        if !self.bic_context_inputs.is_empty() {
             // TODO https://github.com/iotaledger/iota-sdk/issues/1740
             log::debug!("Adding commitment context input for output with block issuance credit context input");
             needs_commitment_context = true;
         }
 
-        if needs_commitment_context
-            && !self
-                .context_inputs
-                .iter()
-                .any(|c| c.kind() == CommitmentContextInput::KIND)
-        {
+        if needs_commitment_context && self.commitment_context_input.is_none() {
             // TODO https://github.com/iotaledger/iota-sdk/issues/1740
-            self.context_inputs
-                .insert(CommitmentContextInput::new(self.latest_slot_commitment_id).into());
+            self.commitment_context_input
+                .replace(CommitmentContextInput::new(self.latest_slot_commitment_id).into());
```

Comment:
Doesn't really make any difference but `replace` while we're under an `is_none` condition feels a bit weird

Reply:
just less verbose